### PR TITLE
[JENKINS-37566] - Stop nullifying channel in ProxyOutputStream#finalize()

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Just YAGNI, it does not help garbage collector much and actually makes it more complicated due to breaking the object chain.

@reviewbybees

### CODE REVIEW

File: src/main/java/hudson/remoting/ProxyOutputStream.java [hunk 1]
```diff
@@ -183,7 +183,6 @@ protected void finalize() throws Throwable {
         // if the object is auto-unexported, the export entry could have already been removed.
         if(channel!=null) {
```
Comment (stephenc) [inline src/main/java/hudson/remoting/ProxyOutputStream.java]: `&& oid != -1`

Comment (oleg-nenashev) [reply src/main/java/hudson/remoting/ProxyOutputStream.java]: yep though there is another issue. Not sure I want to touch it

Comment (stephenc) [reply src/main/java/hudson/remoting/ProxyOutputStream.java]: Well the null assignment was preventing a double export...

Comment (oleg-nenashev) [reply src/main/java/hudson/remoting/ProxyOutputStream.java]: double unexport is actually fine (excepting logging). But OK, will add a check

**Pull request merged**